pkg/provider/run/events.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2021 Nitric Pty Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package run
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/nitrictech/nitric/pkg/plugins/errors"
+	"github.com/nitrictech/nitric/pkg/plugins/errors/codes"
+	"github.com/nitrictech/nitric/pkg/plugins/events"
+	"github.com/nitrictech/nitric/pkg/triggers"
+	"github.com/nitrictech/nitric/pkg/worker"
+)
+
+type WorkerPoolEventService struct {
+	events.UnimplementedeventsPlugin
+	pool worker.WorkerPool
+}
+
+// Publish a message to a given topic
+func (s *WorkerPoolEventService) Publish(topic string, event *events.NitricEvent) error {
+	newErr := errors.ErrorsWithScope(
+		"WorkerPoolEventService.Publish",
+		map[string]interface{}{
+			"topic": topic,
+			"event": event,
+		},
+	)
+
+	requestId := event.ID
+	// payloadType := event.PayloadType
+	payload := event.Payload
+
+	marshaledPayload, err := json.Marshal(payload)
+
```

Comment:
can we not leave a space between the point of error creation and it's check?
This just makes it clearer what the check is associated with, and doesn't make the code quite so spread out.
like so :
```
x, err = foo()
if err != nil {
    return err
}
```